x-pack/plugins/maps/public/classes/layers/vector_layer/vector_layer.tsx

```diff
@@ -757,19 +756,25 @@ export class VectorLayer extends AbstractLayer {
     if (this.isLoadingBounds()) {
       return;
     }
-    await this._syncSourceStyleMeta(syncContext, source, style);
-    await this._syncSourceFormatters(syncContext, source, style);
-    const sourceResult = await this._syncSource(syncContext, source, style);
-    if (
-      !sourceResult.featureCollection ||
-      !sourceResult.featureCollection.features.length ||
-      !this.hasJoins()
-    ) {
-      return;
-    }
 
-    const joinStates = await this._syncJoins(syncContext, style);
-    await this._performInnerJoins(sourceResult, joinStates, syncContext.updateSourceData);
+    try {
+      await this._syncSourceStyleMeta(syncContext, source, style);
+      await this._syncSourceFormatters(syncContext, source, style);
+      const sourceResult = await this._syncSource(syncContext, source, style);
+      if (
+        !sourceResult.featureCollection ||
+        !sourceResult.featureCollection.features.length ||
+        !this.hasJoins()
+      ) {
+        return;
+      }
+
+      const joinStates = await this._syncJoins(syncContext, style);
+      this._performInnerJoins(sourceResult, joinStates, syncContext.updateSourceData);
+    } catch (error) {
```

Comment:
consider checking if this is an abort_sync_data error. If it is not, still throw. Otherwise, there's a risk this is swallowing errors that should not be swallowed.